src/main/java/com/dezorganizacja/bridge/view/graphical/Controller.java

```diff
@@ -0,0 +1,11 @@
+package com.dezorganizacja.bridge.view.graphical;
+
+import com.dezorganizacja.bridge.presenter.MainPresenter;
+
+public class Controller {
+    protected MainPresenter presenter;
```

Comment:
Obawiam się, że nie potrafię odpowiedzieć na Twoje pytanie wprost. To trochę zależy jak rozumiesz "aktywność" prezentera.

Jeżeli chcesz sprawdzić, czy aktualnie jesteś w menu głównym czy w trakcie rozgrywki to owszem, możesz o to odpytać MainPresentera. W zasadzie nie zostało ustalone w jaki sposób możesz go odpytać, więc możesz wymyślić co Ci pasuje (mając tylko troszkę na uwadze co może być potrzebne w przyszłości). Może to być aktualny stan (menu główne/gra/info o autorach/jakikolwiek inny shit), może to być pobranie "aktualnego prezentera" (cokolwiek przez to rozumiesz).

Problem pojawia się nieco potem, jak już jesteś w grze. Ciężko tutaj zdefiniować "aktywny prezenter". W jednym widoku może być kilka prezenterów: 4 prezentery graczy, prezenter stołu, prezenter granego kontraktu... Analogicznie, prawdopodobnie w jednym okienku możesz mieć kilka mniejszych, reużywalnych widoków (na przykład widok kart gracza) które mają swoje kontrolery. Jest to szczególnie prawdziwe, jeżeli chcemy mieć zagnieżdżone widoku (bo menu, bo widok aktualnego kontraktu wewnątrz widoku z grą...)

Nie wiem czy to jest dostatecznie jasne: prezentery **nie prezentują** kompletnej informacji z punktu widzenia interfejsu użytkownika. Prezentery prezentują jedynie małą cząstkę logiki gry (czyli na przykład `PlayerPresenter` odpowiada za zarządzanie kartami gracza).

Żeby zaciemnić Ci obraz jeszcze bardziej ( :wink: ), prawdopodobnie nawet nie potrzebujesz trzymać nic związanego z aktualnym stanem (w rozumieniu aktywnego widoku) w prezenterze. Przykład: klikając przycisk "nowa gra" wywołasz akcję `newGame` na `MainPresenterze`. On z kolei stworzy sobie nowy obiekt ze stanem gry, po czym odpowie Ci `GamePresenterem`. Ty to zobaczysz i stwierdzisz: "o, fajnie, to ja teraz buduję wewnątrz okienka (być może to jest jakiś zagnieżdżony widok - patrz menu) `GameController` i przekazuję mu tylko i wyłacznie prezenter gry. Dalej, `GameController` prawdopodobnie będzie chciał wyświetlić 4 ręce graczy, no ale nie musisz tego pisać 4 razy. Możesz mieć jakiś prezenter, kontroler i widok dla pojedynczej ręki, a potem je złożyć do kupy 4 razy i wykorzystać w `GameControllerze`.